### PR TITLE
feat: add estimate gas bindings

### PR DESCRIPTION
**Pull request by lszetg**
## Motivation and Context

We need to publish gas estimation support for the bindings (previously available only in rust)

## Summary

- added bindings for gas estimation (Swift, WASM, Java)
- changed the default network for examples to iota rebased
- commented out the examples that require backend support

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: bindings/swift/src/ffi_functions.rs [hunk 1]
```diff
@@ -898,6 +898,41 @@ impl ETOPaySdk {
...
+                let amount = CryptoAmount::try_from(amount)?;
+                let pin = EncryptionPin::try_from_string(pin)?;
+                sdk.estimate_gas(&pin, &address, amount, data).await
+            }
+            .await
+            .map(Into::into)
+            .map_err(|err| format!("{:#?}", err))
+        }
+        .await
+        .map_err(|err| format!("{:#?}", err))
```
Comment (antbern) [inline bindings/swift/src/ffi_functions.rs:933]: Can we not remove one level of `async move` here?
```suggestion
        async move {
            let amount = CryptoAmount::try_from(amount)?;
            let pin = EncryptionPin::try_from_string(pin)?;
            sdk.estimate_gas(&pin, &address, amount, data).await
        }
        .await
        .map(Into::into)
        .map_err(|err| format!("{:#?}", err))
```

**Review by antbern — COMMENTED** [review]
Looking good! just one small comment 💯

**Review by kristifidani — COMMENTED** [review]
Overall looks good just 1 comment👍  

Nice to know that Android and Rust examples work.
For swift we can have a look into it on Monday on the Mac.

File: bindings/swift/src/ffi_functions.rs (hunk 1 above)
Comment (lszetg) [reply bindings/swift/src/ffi_functions.rs:933]: good one :D

File: bindings/wasm/bun.lock [hunk 2]
```diff
@@ -0,0 +1,115 @@
+{
+  "lockfileVersion": 1,
+  "workspaces": {
+    "": {
+      "name": "etopay-sdk-wasm-tests",
+      "dependencies": {
```
[the remark this answers is not in the document]
Comment (lszetg) [reply bindings/wasm/bun.lock:6]: probably I did `git add .` once and it happened 🙈

**Review by antbern — APPROVED** [review]

File: bindings/wasm/bun.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (lszetg) [reply bindings/wasm/bun.lock]: ok, it's no longer with us :D

**Review by kristifidani — APPROVED** [review]

**Comment by kristifidani** [comment]
Alright lets go 🚀
